spec/unit/packagers/solaris_spec.rb

```diff
@@ -0,0 +1,231 @@
+            contents = File.read(script_file)
+            expect(contents).to include("Contents of #{script_name}")
+          end
+        end
+      end
+    end
+
+    describe '#write_prototype_file' do
+      let(:prototype_file) { File.join(staging_dir, 'Prototype') }
+
+      before do
+        allow(subject).to receive(:shellout!)
+      end
+
+      it 'creates the prototype file' do
+        subject.write_prototype_file
+        contents = File.read(prototype_file)
+
+        expect(contents).to include(
+          <<-EOH.gsub(/^ {12}/, '')
+            i pkginfo
+            i postinstall
+            i postremove
+          EOH
+        )
+      end
+
+      it 'uses the correct commands' do
+        expect(subject).to receive(:shellout!)
+          .with("cd /opt && find project -print > #{File.join(staging_dir, 'files')}")
+        expect(subject).to receive(:shellout!)
+          .with("cd /opt && pkgproto < #{File.join(staging_dir, 'files')} > #{File.join(staging_dir, 'Prototype.files')}")
+        expect(subject).to receive(:shellout!)
+          .with("awk '{ $5 = \"root\"; $6 = \"root\"; print }' < #{File.join(staging_dir, 'Prototype.files')} >> #{File.join(staging_dir, 'Prototype')}")
+
+        subject.write_prototype_file
+      end
+    end
+
+    describe '#create_solaris_file' do
+      it 'uses the correct commands' do
+        expect(subject).to receive(:shellout!)
+          .with("pkgmk -o -r /opt -d #{staging_dir} -f #{File.join(staging_dir, 'Prototype')}")
+        expect(subject).to receive(:shellout!)
+          .with("pkgchk -vd #{staging_dir} project")
+        expect(subject).to receive(:shellout!)
+          .with("pkgtrans #{staging_dir} #{File.join(package_dir, 'project-1.2.3-1.i386.solaris')} project")
+
+        subject.create_solaris_file
+      end
+    end
+
+    describe '#write_pkginfo_file' do
+      let(:pkginfo_file) { File.join(staging_dir, 'pkginfo') }
+
+      it 'generates the file' do
+        subject.write_pkginfo_file
+        expect(pkginfo_file).to be_a_file
+      end
+
+      it 'has the correct content' do
+        subject.write_pkginfo_file
+        contents = File.read(pkginfo_file)
+
+        expect(contents).to include("CLASSES=none")
+        expect(contents).to include("TZ=PST")
+        expect(contents).to include("PATH=/sbin:/usr/sbin:/usr/bin:/usr/sadm/install/bin")
+        expect(contents).to include("BASEDIR=/opt")
+        expect(contents).to include("PKG=project")
+        expect(contents).to include("NAME=project")
+        expect(contents).to include("ARCH=i386")
+        expect(contents).to include("VERSION=1.2.3-1")
+        expect(contents).to include("CATEGORY=application")
+        expect(contents).to include("DESC=")
+        expect(contents).to include("VENDOR=Chef Software")
+        expect(contents).to include("EMAIL=Chef Software")
+        expect(contents).to include("PSTAMP=#{`hostname`.chomp + Time.now.utc.iso8601}")
```

Comment:
You probably want to stub Time here, travis executed the test on a second boundary and failed.